commit 005411c3e9147bc3b78215390e847d688dbbc163
Author: David Howells <dhowells@redhat.com>
Date:   Tue Jun 16 21:36:49 2009 +0100

    AFS: Correctly translate auth error aborts and don't failover in such cases
    
    Authentication error abort codes should be translated to appropriate
    Linux error codes, rather than all being translated to EREMOTEIO - which
    indicates that the server had internal problems.
    
    Additionally, a server shouldn't be marked unavailable and the next
    server tried if an authentication error occurs.  This will quickly make
    all the servers unavailable to the client.  Instead the error should be
    returned straight to the user.
    
    Signed-off-by: David Howells <dhowells@redhat.com>
    Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>

diff --git a/fs/afs/vlocation.c b/fs/afs/vlocation.c
index ec2a7431e458..6e689208def2 100644
--- a/fs/afs/vlocation.c
+++ b/fs/afs/vlocation.c
@@ -65,6 +65,8 @@ static int afs_vlocation_access_vl_by_name(struct afs_vlocation *vl,
 				goto out;
 			goto rotate;
 		case -ENOMEDIUM:
+		case -EKEYREJECTED:
+		case -EKEYEXPIRED:
 			goto out;
 		default:
 			ret = -EIO;